### PR TITLE
Allow viewing summary for a selected row

### PR DESCRIPTION
**Pull request by riwu**
Allows clicking on a row to view the contribution statistics of the author associated with the row.
Pressing the escape key will return to the summary.

### CODE REVIEW

**Comment by damithc** [comment]
@sebastianquek for your review.

File: src/main/java/gui/MainApp.java [hunk 1]
```diff
@@ -74,11 +75,12 @@ private void initLogic() {
     }
 
     private void addCommandBar(MainApp mainApp) {
-        rootLayout.setBottom(new CommandBarController(mainApp));
+    	commandBarController = new CommandBarController(mainApp);
```
Comment (sebastianquek) [inline src/main/java/gui/MainApp.java]: Whitespace formatting

File: src/main/java/gui/MainApp.java [hunk 2]
```diff
@@ -91,16 +93,15 @@ private void addFileStats(String authorName) {
...
 
-    public void handleKeyPress(CommandBarController commandBarController,
-                               KeyCode key,
-                               String userInput) {
+    public void handleKeyPress(KeyCode key, String userInput) {
         if (key == KeyCode.ENTER) {
-            handleEnterPress(commandBarController, userInput);
+            handleEnterPress(userInput);
+        } else if (key == KeyCode.ESCAPE) {
+        	addSummary(this);
```
Comment (sebastianquek) [inline src/main/java/gui/MainApp.java]: Use 4 spaces to indent, not tabs

File: src/main/java/gui/MainApp.java [hunk 3]
```diff
@@ -127,4 +128,11 @@ private void handleEnterPress(CommandBarController commandBarController,
         }
         commandBarController.clear();
     }
+    
+    public void handleMouseClick(AuthorBean selectedAuthor) {
+    	String authorName = selectedAuthor.nameProperty().get();
```
Comment (sebastianquek) [inline src/main/java/gui/MainApp.java]: Whitespace formatting, this includes the several lines after this.

File: src/main/java/gui/SummaryController.java [hunk 4]
```diff
@@ -33,7 +33,7 @@
 
     private static final String OVERVIEW_LAYOUT_FXML = "/main/resources/layouts/Summary.fxml";
 
-    public SummaryController(Collection<Author> inputSummaryData) {
+	public SummaryController(MainApp mainApp, Collection<Author> inputSummaryData) {
```
Comment (sebastianquek) [inline src/main/java/gui/SummaryController.java]: Use 4 spaces

File: src/main/java/gui/SummaryController.java [hunk 5]
```diff
@@ -50,6 +50,10 @@ public SummaryController(Collection<Author> inputSummaryData) {
         }
 
         summaryTable.setItems(summaryData);
+        summaryTable.getSelectionModel().selectedItemProperty().addListener((observableValue, oldValue, newValue) -> {
+        	mainApp.handleMouseClick(newValue);
```
Comment (sebastianquek) [inline src/main/java/gui/SummaryController.java]: Use 4 spaces

**Review by sebastianquek — CHANGES_REQUESTED** [review]
Overall, looks good! 👍 

Do take note of the formatting, I've followed the Java Coding Standards as mentioned in CS2103.

File: src/main/java/gui/MainApp.java (hunk 1 above)
Comment (riwu) [reply src/main/java/gui/MainApp.java]: What's whitespace formatting? Is it a different problem from tab indentation?

Also do you indent empty lines?

Comment (sebastianquek) [reply src/main/java/gui/MainApp.java]: It's the same as tab indentation, use 4 spaces.
No, do not indent empty lines!

**Comment by riwu** [comment]
Fixed the tab indentation with autoformatting, which removed a few pre-existing empty line indentation as well, let me know if I missed anything.
Do you want the commits to be squashed into one btw?

File: src/main/java/gui/SummaryController.java [hunk 6]
```diff
@@ -43,13 +43,18 @@ public SummaryController(Collection<Author> inputSummaryData) {
...
-        
+
         ObservableList<AuthorBean> summaryData = FXCollections.observableArrayList();
         for (Author author : inputSummaryData) {
             summaryData.add(new AuthorBean(author));
         }
 
         summaryTable.setItems(summaryData);
+        summaryTable.getSelectionModel().selectedItemProperty()
+                .addListener((observableValue, oldValue, newValue) -> {
```
Comment (sebastianquek) [inline src/main/java/gui/SummaryController.java]: Could you also indent this block to match the previous statement's dot separator?

**Comment by sebastianquek** [comment]
@riwu Sure, you can squash the commits into one! Thanks!!

**Review by sebastianquek — APPROVED** [review]

**Pull request merged**